ios/objective-c/AmpliObjectiveCSampleApp/AmpliObjectiveCSampleApp/Ampli/Ampli.m

```diff
@@ -35,1213 +35,422 @@ - (instancetype)initWithEventType: (NSString *) eventType withEventProperties: (
 }
 
-- (instancetype)initWithValue:(NSString *)value
-{
-    if (self = [super init]) _value = value;
+- (instancetype)initWithRequiredNumber_Identify:(Float64)requiredNumber
+optionalArray:(NSArray<NSString *> * _Nullable)optionalArray {
+    self = [super initWithEventType:@"Identify"
+                    withEventProperties:@{
+                        @"optionalArray": optionalArray ?: NSNull.null,
+                        @"requiredNumber": @(requiredNumber)
+                    }];
     return self;
 }
-
-- (NSUInteger)hash { return _value.hash; }
 @end
 
-@implementation EventWithDifferentCasingTypesEnumWithSpace
-+ (NSDictionary<NSString *, EventWithDifferentCasingTypesEnumWithSpace *> *)values
-{
-    static NSDictionary<NSString *, EventWithDifferentCasingTypesEnumWithSpace *> *values;
-    return values = values ? values : @{
-        @"enum with space": [[EventWithDifferentCasingTypesEnumWithSpace alloc] initWithValue:@"enum with space"],
-    };
-}
+#pragma mark - EventMaxIntForTest
 
-+ (EventWithDifferentCasingTypesEnumWithSpace *)enumWithSpace { return EventWithDifferentCasingTypesEnumWithSpace.values[@"enum with space"]; }
+@implementation EventMaxIntForTest: Event
 
-+ (instancetype _Nullable)withValue:(NSString *)value
-{
-    return EventWithDifferentCasingTypesEnumWithSpace.values[value];
++ (instancetype)intMax10:(NSInteger)intMax10 {
+    return [[self alloc] initWithIntMax10_EventMaxIntForTest: intMax10];
 }
 
-- (instancetype)initWithValue:(NSString *)value
-{
-    if (self = [super init]) _value = value;
+- (instancetype)initWithIntMax10_EventMaxIntForTest:(NSInteger)intMax10 {
+    self = [super initWithEventType:@"EventMaxIntForTest"
+                    withEventProperties:@{
+                        @"intMax10": @(intMax10)
+                    }];
     return self;
 }
-
-- (NSUInteger)hash { return _value.hash; }
 @end
 
-@implementation EventWithEnumTypesOptionalEnum
-+ (NSDictionary<NSString *, EventWithEnumTypesOptionalEnum *> *)values
-{
-    static NSDictionary<NSString *, EventWithEnumTypesOptionalEnum *> *values;
-    return values = values ? values : @{
-        @"optional enum 1": [[EventWithEnumTypesOptionalEnum alloc] initWithValue:@"optional enum 1"],
-        @"optional enum 2": [[EventWithEnumTypesOptionalEnum alloc] initWithValue:@"optional enum 2"],
-    };
-}
-
-+ (EventWithEnumTypesOptionalEnum *)optionalEnum1 { return EventWithEnumTypesOptionalEnum.values[@"optional enum 1"]; }
-+ (EventWithEnumTypesOptionalEnum *)optionalEnum2 { return EventWithEnumTypesOptionalEnum.values[@"optional enum 2"]; }
-
-+ (instancetype _Nullable)withValue:(NSString *)value
-{
-    return EventWithEnumTypesOptionalEnum.values[value];
-}
+#pragma mark - EventNoProperties
 
-- (instancetype)initWithValue:(NSString *)value
-{
-    if (self = [super init]) _value = value;
+@implementation EventNoProperties: Event
+- (instancetype)init {
+    self = (EventNoProperties *) [Event withEventType:@"Event No Properties"
```

Comment:
@falconandy I was getting errors with your original code. I had to switch to `[EventNoProperties new]` in the Event method. Not sure if this is the best fix but it got things working.

Reply:
Fixed